### PR TITLE
Add support for VK_EXT_external_memory_host extension.

### PR DESCRIPTION
**Pull request by billhollings**
- To expose host memory page size, add `mvkGetHostMemoryPageSize()` and `MVKPhysicalDeviceMetalFeatures::hostMemoryPageSize`.
- `MVKPhysicalDevice::getProperties()` sort EXT property structs alphabetically (unrelated).
- `MVKExtensions.def` fix alignment in OS versions (unrelated).

Implements #794.

### CODE REVIEW

File: MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm [hunk 1]
```diff
@@ -3469,6 +3483,27 @@ static uint32_t mvkGetEntryProperty(io_registry_entry_t entry, CFStringRef prope
...
+	if (pMemHostPtrProps) {
+		switch (handleType) {
+			case VK_EXTERNAL_MEMORY_HANDLE_TYPE_HOST_ALLOCATION_BIT_EXT:
+				pMemHostPtrProps->memoryTypeBits = _physicalDevice->getHostVisibleMemoryTypes();
+				break;
+			case VK_EXTERNAL_MEMORY_HANDLE_TYPE_HOST_MAPPED_FOREIGN_MEMORY_BIT_EXT:
+				// Vulkan requires only host-visible memory types here, but Metal
+				// only supports Private storage for cross-device buffer sharing.
+				pMemHostPtrProps->memoryTypeBits = 0;
+				break;
```
Comment (cdavis5e) [inline MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm:3498]: AFAICT, the docs for `-[MTLDevice newBufferWithBytesNoCopy:length:options:deallocator:]` only state that the memory allocation must belong to a single VM region. I don't see how passing the `-[MTLBuffer contents]` of a foreign device's buffer violates that constraint. What happens when you do that?

Also, I don't recall Metal allowing the direct sharing of `MTLBuffer` objects at all. All of the resource-sharing mechanisms I know of (`MTLSharedTextureHandle`, `IOSurface`) only publicly support `MTLTexture`s. (I know about the secret `MTLDevice` methods for creating `MTLBuffer`s from `IOSurface`s, but they're obviously not supported for public use.) The one thing I think you might be thinking of, I believe is intended to be used within a peer group (i.e. a Vulkan device group).

**Review by cdavis5e — CHANGES_REQUESTED** [review]
Looks good. I just have one objection...

File: MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm (hunk 1 above)
Comment (billhollings) [reply MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm]: > I don't see how passing the `-[MTLBuffer contents]` of a foreign device's buffer violates that constraint.

Ah. Good insight! Thanks.

I agree host-mapped memory allocated on another GPU should be fine to import. I've made changes to support `VK_EXTERNAL_MEMORY_HANDLE_TYPE_HOST_MAPPED_FOREIGN_MEMORY_BIT_EXT `.

>  I don't recall Metal allowing the direct sharing of MTLBuffer objects at all.

macOS Metal supports creating a `MTLBuffer` as a view on a foreign `MTLBuffer`, but only for `Private` storage and BLIT operations, so not useful for this extension:

- [-`[MTBuffer newRemoteBufferViewForDevice:]`](https://developer.apple.com/documentation/metal/mtlbuffer/2967415-makeremotebufferview)
- [-`[MTBuffer remoteStorageBuffer]`](https://developer.apple.com/documentation/metal/mtlbuffer/2967416-remotestoragebuffer)

Comment (cdavis5e) [reply MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm]: > macOS Metal supports creating a `MTLBuffer` as a view on a foreign `MTLBuffer`, but only for `Private` storage and BLIT operations, so not useful for this extension:
> 
> * [-`[MTBuffer newRemoteBufferViewForDevice:]`](https://developer.apple.com/documentation/metal/mtlbuffer/2967415-makeremotebufferview)
> * [-`[MTBuffer remoteStorageBuffer]`](https://developer.apple.com/documentation/metal/mtlbuffer/2967416-remotestoragebuffer)

Yep, that's the one I thought you were thinking of. The docs also state that this device and the other device must belong to the same peer group. We'd only need it to implement device groups for real.

**Review by cdavis5e — APPROVED** [review]

**Comment by DUOLabs333** [comment]
Am I doing something wrong? I tried enabling this with `ext_names[ext_count++] = VK_KHR_EXTERNAL_MEMORY_EXTENSION_NAME;`, but I get `VK_ERROR_EXTENSION_NOT_PRESENT`.

**Comment by DUOLabs333** [comment]
Am I missing something? I tried enabling it with 
```
ext_names[ext_count++] = VK_EXT_EXTERNAL_MEMORY_HOST_EXTENSION_NAME;
```
but I got `VK_ERROR_EXTENSION_NOT_PRESENT`.

**Pull request merged**